userfiles/minc_file/views/_volume_viewer.html.erb

```diff
@@ -743,15 +743,41 @@
         $("#voxel-k-<%= bb_identifier %>-" + vol_id).val(parseInt(voxel_coords.k, 10));
       }
 
+      /* Set background color of intensity value to match colormap
+      * entry for that value.
+      */
       value = volume.getIntensityValue();
-      $("#intensity-value-<%= bb_identifier %>-" + vol_id)
-      .css("background-color", "#" + volume.color_map.colorFromValue(value, {
+      var bg_color = volume.color_map.colorFromValue(value, {
         hex: true,
-        min: volume.min,
-        max: volume.max,
+        min: volume.intensity_min,
+        max: volume.intensity_max,
         contrast: panel.contrast,
         brightness: panel.brightness
-      }))
+      });
+
+      /* This function simply takes an input hex background color
+      * and returns either "black" or "white" as the appropriate
+      * foreground color for text rendered over the background colour.
+      * Idea from https://24ways.org/2010/calculating-color-contrast/
+      * Equation is from http://www.w3.org/TR/AERT#color-contrast
+      */
+      function getContrastYIQ(hexcolor) {
```

Comment:
Why is this function smack in the middle of the event handler? Every time the sliceupdate even is triggered, this function will be redefined again and again.

The code looks like

```
viewer.addEventListener("sliceupdate"... {
  value=...
  var bg_color=...
  function getContrastYIQ {}
  var fg_color=getContrastYIQ()
  (set css here)
} // end of handler
```